Client/multiplayer_sa/CMultiplayerSA_FrameRateFixes.cpp

```diff
@@ -74,6 +73,134 @@ static void _declspec(naked) HOOK_CHeli__ProcessFlyingCarStuff()
+        fstp [esp+0x28]
+        fld [esp+0x24]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fstp [esp+0x24]
+        fld [esp+0x20]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fadd [esi]
+        jmp RETURN_CFallingGlassPane__Update_A
+    }
+}
+
+#define HOOKPOS_CFallingGlassPane__Update_B 0x71AAEA
+#define HOOKSIZE_CFallingGlassPane__Update_B 0x6
+static const unsigned int RETURN_CFallingGlassPane__Update_B = 0x71AAF0;
+static void _declspec(naked) HOOK_CFallingGlassPane__Update_B()
+{
+    _asm {
+        fld [eax]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fstp [eax]
+        fld [eax+0x4]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fstp [eax+0x4]
+        fld [eax+0x8]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fstp [eax+0x8]
+        mov ecx, [eax]
+        mov [esp+0x2C], ecx
+        jmp RETURN_CFallingGlassPane__Update_B
+    }
+}
+
+#define HOOKPOS_CFallingGlassPane__Update_C 0x71AB29
+#define HOOKSIZE_CFallingGlassPane__Update_C 0x6
+static const unsigned int RETURN_CFallingGlassPane__Update_C = 0x71AB2F;
+static void _declspec(naked) HOOK_CFallingGlassPane__Update_C()
+{
+    _asm {
+        fld [eax]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fstp [eax]
+        fld [eax+0x4]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fstp [eax+0x4]
+        fld [eax+0x8]
+        fmul ds:[0xB7CB5C]          // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        fstp [eax+0x8]
+        mov edx, [eax]
+        mov [esp+0x38], edx
+        jmp RETURN_CFallingGlassPane__Update_C
+    }
+}
+
+#define HOOKPOS_CPhysical__ApplyAirResistance 0x544D29
+#define HOOKSIZE_CPhysical__ApplyAirResistance 5
+static const unsigned int RETURN_CPhysical__ApplyAirResistance = 0x544D4D;
+static void _declspec(naked) HOOK_CPhysical__ApplyAirResistance()
+{
+    _asm {
+        fld ds:[0x862CD0]           // 0.99000001f
+        fld ds:[0xB7CB5C]           // CTimer::ms_fTimeStep
+        fdiv kOriginalTimeStep      // 1.666f
+        mov eax, 0x822130           // powf
+        call eax
+
+        fld st(0)
+        fmul [esi+0x50]
+        fstp [esi+0x50]
+
+        fld st(0)
+        fmul [esi+0x54]
+        fstp [esi+0x54]
+
+        fmul [esi+0x58]
+        fstp [esi+0x58]
+        jmp RETURN_CPhysical__ApplyAirResistance
+    }
+}
+
+static unsigned int RETURN_VehicleRapidStopFix;
+template <unsigned int returnAddress>
+static void _declspec(naked) HOOK_VehicleRapidStopFix()
+{
+    RETURN_VehicleRapidStopFix = returnAddress;
```

Comment:
Wouldn't it make more sense to turn this line into a static variable plus assignment?

Reply:
Changed it. For some reason I thought that you can not have a static variable in a naked function. But apperently you can. I also checked the assembly of release and debug build and it looks fine.